client/dashboard/profile-wizard/steps/business-details.js

```diff
@@ -119,14 +119,17 @@ class BusinessDetails extends Component {
 	}
 
 	getNumberRangeString( min, max = false, format = numberFormat ) {
+		// The localized strings below do not use sprintf to allow
+		// formatting the range on a per currency basis via translation.
 		if ( ! max ) {
-			return sprintf( _x( '%s+', 'store product count', 'woocommerce-admin' ), format( min ) );
+			return _x( format( min ) + '+', 'store product count or revenue', 'woocommerce-admin' );
 		}
 
-		return sprintf(
-			_x( '%s - %s', 'store product count', 'woocommerce-admin' ),
```

Comment:
If this gets restored, it should just positional formatting so RTL languages can swap the values. E.g. `%1$s - %2$s`.